satpy/_config.py

```diff
@@ -0,0 +1,136 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# Copyright (c) 2016-2019 Satpy developers
+#
+# This file is part of satpy.
+#
+# satpy is free software: you can redistribute it and/or modify it under the
+# terms of the GNU General Public License as published by the Free Software
+# Foundation, either version 3 of the License, or (at your option) any later
+# version.
+#
+# satpy is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+# A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# satpy.  If not, see <http://www.gnu.org/licenses/>.
+"""Satpy Configuration directory and file handling."""
+
+import glob
+import logging
+import os
+import sys
+from collections import OrderedDict
+
+import pkg_resources
+from donfig import Config
+import appdirs
+
+LOG = logging.getLogger(__name__)
+
+BASE_PATH = os.path.dirname(os.path.realpath(__file__))
+# FIXME: Use package_resources?
+PACKAGE_CONFIG_PATH = os.path.join(BASE_PATH, 'etc')
+
+_satpy_dirs = appdirs.AppDirs(appname='satpy', appauthor='pytroll')
+_CONFIG_DEFAULTS = {
+    'cache_dir': _satpy_dirs.user_cache_dir,
+    'data_dir': _satpy_dirs.user_data_dir,
+    'config_path': [],
+}
+
+# Satpy main configuration object
+# See https://donfig.readthedocs.io/en/latest/configuration.html
+# for more information.
+#
+# Configuration values will be loaded from files at:
+# 1. The builtin package satpy.yaml (not present currently)
+# 2. $SATPY_ROOT_CONFIG (default: /etc/satpy/satpy.yaml)
+# 3. <python-env-prefix>/etc/satpy/satpy.yaml
+# 4. ~/.config/satpy/satpy.yaml
+# 5. ~/.satpy/satpy.yaml
+# 6. $SATPY_CONFIG_PATH/satpy.yaml if present (colon separated)
+_CONFIG_PATHS = [
+    os.path.join(PACKAGE_CONFIG_PATH, 'satpy.yaml'),
+    os.getenv('SATPY_ROOT_CONFIG', os.path.join('/etc', 'satpy', 'satpy.yaml')),
+    os.path.join(sys.prefix, 'etc', 'satpy', 'satpy.yaml'),
+    os.path.join(os.path.expanduser('~'), '.config', 'satpy', 'satpy.yaml'),
+    os.path.join(os.path.expanduser('~'), '.satpy', 'satpy.yaml'),
+]
+# The above files can also be directories. If directories all files
+# with `.yaml`., `.yml`, or `.json` extensions will be used.
+
+_ppp_config_dir = os.getenv('PPP_CONFIG_DIR', None)
+_satpy_config_path = os.getenv('SATPY_CONFIG_PATH', None)
+if _ppp_config_dir is not None and _satpy_config_path is None:
+    LOG.warning("'PPP_CONFIG_DIR' is deprecated. Please use 'SATPY_CONFIG_PATH' instead.")
+    _satpy_config_path = _ppp_config_dir
+    os.environ['SATPY_CONFIG_PATH'] = _satpy_config_path
```

Comment:
Don't we want to print the warning also when SATPY_CONFIG_DIR is specified? this would show the user that there is still a problem...

Reply:
Isn't PPP_CONFIG_DIR still used by some other packages? pyspectral? pyresample? My worry was that someone would have PPP defined for those packages and they would always get this warning even though they were properly using SATPY_CONFIG_PATH.

Reply:
I can't see any use of PPP_CONFIG_DIR in pyresample or pyspectral...

Reply:
Ah I was thinking about pyspectral's configuration file variable: https://pyspectral.readthedocs.io/en/master/installation.html#configuration-file

Ok, well then if we assume that `PPP_CONFIG_DIR` is not used by any other pytroll package then I will change this block of code to always print the warning.

Reply:
Done.